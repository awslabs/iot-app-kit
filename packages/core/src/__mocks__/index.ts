export * from './asset';
export * from './assetModel';
export * from './assetPropertyValue';
export * from './iotsitewiseSDK';
export * from '../../../core/src/mockWidgetProperties';
export * from './data-source';
