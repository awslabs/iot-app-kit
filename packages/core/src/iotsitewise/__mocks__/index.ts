export * from './asset';
export * from './assetModel';
export * from './assetPropertyValue';
export * from './iotsitewiseSDK';
export * from './mockWidgetProperties';
export * from './data-source';
