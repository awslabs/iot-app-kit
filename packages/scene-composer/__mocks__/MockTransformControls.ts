export class MockTransformControls {
  static instance;

  object: any = {};
  showY = true;
  flipY = false;
  camera;
  domElement;

  constructor(camera, domElement) {
    this.camera = camera;
    this.domElement = domElement;

    MockTransformControls.instance = this;
  }

  static addEventListener = vi.fn();
  static removeEventListener = vi.fn();
  static setMode = vi.fn();
  static attach = vi.fn();
  static detach = vi.fn();

  addEventListener = MockTransformControls.addEventListener;
  removeEventListener = MockTransformControls.removeEventListener;
  setMode = MockTransformControls.setMode;
  attach = MockTransformControls.attach;
  detach = MockTransformControls.detach;
  traverse = vi.fn();
}
