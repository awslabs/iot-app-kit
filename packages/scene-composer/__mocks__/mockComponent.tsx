import React from 'react';

export default (componentName) => {
  // eslint-disable-next-line react/prop-types
  const component = ({ children, ...props }) => {
    const snapshotableProps = {};
    const renderProps = [];

    Object.entries(props).forEach(([key, value]) => {
      try {
        // Covers edge case where stringify fails due to circular object graphs
        if (typeof value === 'object') {
          if (React.isValidElement(value)) {
            renderProps[key] = value;
          } else if (key === 'style') {
            snapshotableProps[key] = value;
          } else {
            snapshotableProps[key] = JSON.stringify(value);
          }
        } else {
          snapshotableProps[key] = value;
        }
      } catch (e) {
        snapshotableProps[key] = value;
      }
    });

    return (
      <div data-mocked={componentName} {...snapshotableProps}>
        {Object.entries(renderProps).map(([key, value]) => (
          <div key={key}>{value}</div>
        ))}
        {children}
      </div>
    );
  };

   component.displayName = `Mock${componentName}`;
   return component
};
