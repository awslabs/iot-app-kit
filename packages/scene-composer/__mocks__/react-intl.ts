const strings = require('../translations/IotAppKitSceneComposer.en_US.json');

const reactIntl = jest.requireActual('react-intl');
type Message = {
  note: string;
  text: string;
};

const messages = Object.entries(strings).reduce((acc, [key, msg]: [key: string, msg: any]) => {
  acc[key] = msg.text;
  return acc;
}, {});

const intl = new reactIntl.createIntl({
  locale: 'en',
  messages
});

module.exports = {
  ...reactIntl,
  useIntl: jest.fn(() => intl),
};
