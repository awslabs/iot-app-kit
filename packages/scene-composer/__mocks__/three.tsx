/**
 * Mock implementation of R3F (React Three Fiber) components with the string name for a clean snapshot test.
 */
module.exports = {
  ...jest.requireActual('three'),
  audioListener: 'audioListener',
  positionalAudio: 'positionalAudio',

  mesh: 'mesh',
  instancedMesh: 'instancedMesh',
  scene: 'scene',
  sprite: 'sprite',
  lOD: 'lOD',
  skinnedMesh: 'skinnedMesh',
  skeleton: 'skeleton',
  bone: 'bone',
  lineSegments: 'lineSegments',
  lineLoop: 'lineLoop',
  points: 'points',
  group: 'group',
  immediateRenderObject: 'immediateRenderObject',

  // cameras
  camera: 'camera',
  perspectiveCamera: 'perspectiveCamera',
  orthographicCamera: 'orthographicCamera',
  cubeCamera: 'cubeCamera',
  arrayCamera: 'arrayCamera',

  // geometry
  instancedBufferGeometry: 'instancedBufferGeometry',
  bufferGeometry: 'bufferGeometry',
  boxBufferGeometry: 'boxBufferGeometry',
  circleBufferGeometry: 'circleBufferGeometry',
  coneBufferGeometry: 'coneBufferGeometry',
  cylinderBufferGeometry: 'cylinderBufferGeometry',
  dodecahedronBufferGeometry: 'dodecahedronBufferGeometry',
  extrudeBufferGeometry: 'extrudeBufferGeometry',
  icosahedronBufferGeometry: 'icosahedronBufferGeometry',
  latheBufferGeometry: 'latheBufferGeometry',
  octahedronBufferGeometry: 'octahedronBufferGeometry',
  parametricBufferGeometry: 'parametricBufferGeometry',
  planeBufferGeometry: 'planeBufferGeometry',
  polyhedronBufferGeometry: 'polyhedronBufferGeometry',
  ringBufferGeometry: 'ringBufferGeometry',
  shapeBufferGeometry: 'shapeBufferGeometry',
  sphereBufferGeometry: 'sphereBufferGeometry',
  tetrahedronBufferGeometry: 'tetrahedronBufferGeometry',
  textBufferGeometry: 'textBufferGeometry',
  torusBufferGeometry: 'torusBufferGeometry',
  torusKnotBufferGeometry: 'torusKnotBufferGeometry',
  tubeBufferGeometry: 'tubeBufferGeometry',
  wireframeGeometry: 'wireframeGeometry',
  parametricGeometry: 'parametricGeometry',
  tetrahedronGeometry: 'tetrahedronGeometry',
  octahedronGeometry: 'octahedronGeometry',
  icosahedronGeometry: 'icosahedronGeometry',
  dodecahedronGeometry: 'dodecahedronGeometry',
  polyhedronGeometry: 'polyhedronGeometry',
  tubeGeometry: 'tubeGeometry',
  torusKnotGeometry: 'torusKnotGeometry',
  torusGeometry: 'torusGeometry',
  textGeometry: 'textGeometry',
  sphereGeometry: 'sphereGeometry',
  ringGeometry: 'ringGeometry',
  planeGeometry: 'planeGeometry',
  latheGeometry: 'latheGeometry',
  shapeGeometry: 'shapeGeometry',
  extrudeGeometry: 'extrudeGeometry',
  edgesGeometry: 'edgesGeometry',
  coneGeometry: 'coneGeometry',
  cylinderGeometry: 'cylinderGeometry',
  circleGeometry: 'circleGeometry',
  boxGeometry: 'boxGeometry',

  // materials
  material: 'material',
  shadowMaterial: 'shadowMaterial',
  spriteMaterial: 'spriteMaterial',
  rawShaderMaterial: 'rawShaderMaterial',
  shaderMaterial: 'shaderMaterial',
  pointsMaterial: 'pointsMaterial',
  meshPhysicalMaterial: 'meshPhysicalMaterial',
  meshStandardMaterial: 'meshStandardMaterial',
  meshPhongMaterial: 'meshPhongMaterial',
  meshToonMaterial: 'meshToonMaterial',
  meshNormalMaterial: 'meshNormalMaterial',
  meshLambertMaterial: 'meshLambertMaterial',
  meshDepthMaterial: 'meshDepthMaterial',
  meshDistanceMaterial: 'meshDistanceMaterial',
  meshBasicMaterial: 'meshBasicMaterial',
  meshMatcapMaterial: 'meshMatcapMaterial',
  lineDashedMaterial: 'lineDashedMaterial',
  lineBasicMaterial: 'lineBasicMaterial',

  // primitive
  primitive: 'primitive',

  // lights and other
  light: 'light',
  spotLightShadow: 'spotLightShadow',
  spotLight: 'spotLight',
  pointLight: 'pointLight',
  rectAreaLight: 'rectAreaLight',
  hemisphereLight: 'hemisphereLight',
  directionalLightShadow: 'directionalLightShadow',
  directionalLight: 'directionalLight',
  ambientLight: 'ambientLight',
  lightShadow: 'lightShadow',
  ambientLightProbe: 'ambientLightProbe',
  hemisphereLightProbe: 'hemisphereLightProbe',
  lightProbe: 'lightProbe',

  // helpers
  spotLightHelper: 'spotLightHelper',
  skeletonHelper: 'skeletonHelper',
  pointLightHelper: 'pointLightHelper',
  hemisphereLightHelper: 'hemisphereLightHelper',
  gridHelper: 'gridHelper',
  polarGridHelper: 'polarGridHelper',
  directionalLightHelper: 'directionalLightHelper',
  cameraHelper: 'cameraHelper',
  boxHelper: 'boxHelper',
  box3Helper: 'box3Helper',
  planeHelper: 'planeHelper',
  arrowHelper: 'arrowHelper',
  axesHelper: 'axesHelper',

  // textures
  texture: 'texture',
  videoTexture: 'videoTexture',
  dataTexture: 'dataTexture',
  dataTexture3D: 'dataTexture3D',
  compressedTexture: 'compressedTexture',
  cubeTexture: 'cubeTexture',
  canvasTexture: 'canvasTexture',
  depthTexture: 'depthTexture',

  // misc
  raycaster: 'raycaster',
  vector2: 'vector2',
  vector3: 'vector3',
  vector4: 'vector4',
  euler: 'euler',
  matrix3: 'matrix3',
  matrix4: 'matrix4',
  quaternion: 'quaternion',
  bufferAttribute: 'bufferAttribute',
  instancedBufferAttribute: 'instancedBufferAttribute',
  color: 'color',
  fog: 'fog',
  fogExp2: 'fogExp2',
  shape: 'shape',
};
