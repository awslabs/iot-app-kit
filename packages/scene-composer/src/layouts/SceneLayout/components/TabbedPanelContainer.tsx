import React from 'react';
import { Tabs } from '@awsui/components-react';

type TabbedPanelContainerProps = {
  panels: Record<string, JSX.Element>;
};

type TabbedPanelContainerState = {};

class TabbedPanelContainer extends React.Component<TabbedPanelContainerProps, TabbedPanelContainerState> {
  public static defaultProps = {
    menuBar: null,
    statusBar: null,
    leftPanel: null,
    rightPanel: null,
  };

  render() {
    const tabs = Object.keys(this.props.panels).map((tabName) => {
      return {
        label: tabName,
        id: tabName,
        content: this.props.panels[tabName],
      };
    });
    return <Tabs className='sidePanelTabs' tabs={tabs} />;
  }
}

export default TabbedPanelContainer;
