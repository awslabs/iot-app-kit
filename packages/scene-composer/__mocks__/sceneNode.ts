import { ISceneNodeInternal } from "../src/store";

export const defaultNode: ISceneNodeInternal = {
    ref: 'node-ref',
    name: 'default-node',
    transform: {
        position: [0, 0, 0],
        rotation: [0, 0, 0],
        scale: [1, 1, 1]
    },
    transformConstraint: {},
    childRefs: [],
    components: [],
    properties: {},
};
