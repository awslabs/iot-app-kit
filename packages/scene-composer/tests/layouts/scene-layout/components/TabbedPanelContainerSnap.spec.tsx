/* eslint-disable import/first */
import renderer from 'react-test-renderer';
import React from 'react';

// eslint-disable-next-line import/order
import { mockPolaris } from '../../../__mocks__/MockPolaris';

mockPolaris();

import TabbedPanelContainer from '../../../../src/layouts/scene-layout/components/TabbedPanelContainer';

describe('TabbedPanelContainerSnap', () => {
  it('should render correctly', () => {
    const container = renderer.create(
      <TabbedPanelContainer panels={{ test1: <div> Panel 1 </div>, test2: <div> Panel 2 </div> }} />,
    );
    expect(container).toMatchSnapshot();
  });
});
