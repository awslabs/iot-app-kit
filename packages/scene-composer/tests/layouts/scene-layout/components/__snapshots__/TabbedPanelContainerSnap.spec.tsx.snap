// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabbedPanelContainerSnap should render correctly 1`] = `
<Tabs
  className="sidePanelTabs"
  tabs={
    Array [
      Object {
        "content": <div>
           Panel 1 
        </div>,
        "id": "test1",
        "label": "test1",
      },
      Object {
        "content": <div>
           Panel 2 
        </div>,
        "id": "test2",
        "label": "test2",
      },
    ]
  }
/>
`;
